src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\HttpCache\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Sets the list of resources' IRIs included in this response in the "Cache-Tags" HTTP header.
+ *
+ * The "Cache-Tags" is used because it is supported by CloudFlare.
+ *
+ * @see https://support.cloudflare.com/hc/en-us/articles/206596608-How-to-Purge-Cache-Using-Cache-Tags-Enterprise-only-
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class AddTagsListener
+{
+    private $iriConverter;
+
+    public function __construct(IriConverterInterface $iriConverter)
+    {
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * Adds the "Cache-Tags" header.
+     *
+     * @param FilterResponseEvent $event
+     */
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $response = $event->getResponse();
+
+        if (
+            !$request->isMethodCacheable()
+            || !$response->isCacheable()
+            || (!$attributes = RequestAttributesExtractor::extractAttributes($request))
+            || !$resources = $request->attributes->get('_resources')
```

Comment:
The parenthesis on the previous line are mandatory because of operator priorities.